Signal/src/environment/SignalApp.m

```diff
@@ -55,97 +55,90 @@ - (instancetype)initDefault
 
 #pragma mark - Singletons
 
+- (void)createSingletons
+{
+    OWSAssert(SSKEnvironment.shared.messageSender);
```

Comment:
`SignalApp` creates and owns many singletons that are main-app-only.  We now make them in one go, (rather than lazy-creating them as we did before), with an explicit ordering.